bin/etre/main.go

```diff
@@ -6,29 +6,56 @@ import (
 	"crypto/tls"
 	"crypto/x509"
 	"flag"
+	"fmt"
 	"io/ioutil"
 	"log"
+	"net/http"
 	"os"
 
+	"github.com/square/etre/api"
 	"github.com/square/etre/db"
+	"github.com/square/etre/router"
 
 	"gopkg.in/yaml.v2"
 )
 
-type Config struct {
+type DBConfig struct {
 	URL        string `yaml:"url"`
 	Database   string `yaml:"database"`
 	Collection string `yaml:"collection"`
 	Timeout    int    `yaml:"timeout"`
-	TLSCert    string `yaml:"tls-cert"`
-	TLSKey     string `yaml:"tls-key"`
-	TLSCA      string `yaml:"tls-ca"`
+
+	// Certs
+	TLSCert string `yaml:"tls-cert"`
+	TLSKey  string `yaml:"tls-key"`
+	TLSCA   string `yaml:"tls-ca"`
+
+	// Credentials
+	Username  string `yaml:"username"`
+	Source    string `yaml:"source"`
+	Mechanism string `yaml:"mechanism"`
+}
+
+type ServerConfig struct {
+	Port int `yaml:"port"`
```

Comment:
Let's do `Addr` instead, default `127.0.0.1:8080` (probably another default port is better because 8080 is already commonly used). Just `:port` binds to all interfaces, which is considered insecure. Best practice is binding only to localhost by default.